docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,129 @@
+# Contributing to Kubeflow Pipelines Components
+
+Welcome! This guide covers how to contribute components to the Kubeflow Pipelines Components Repository.
+
+## Table of Contents
+
+- [Quick Start](#quick-start)
+- [What We Accept](#what-we-accept)
+- [Component Structure](#component-structure)
+- [Development Workflow](#development-workflow)
+- [Testing Requirements](#testing-requirements)
+- [Review Process](#review-process)
+- [Before Submitting](#before-submitting)
+- [Getting Help](#getting-help)
+- [Naming Conventions](#naming-conventions)
+- [References](#references)
+
+## Quick Start
+
+*Get up and running with your first contribution in 7 simple steps.*
+
+1. **Read the [Kubeflow Community Code of Conduct](https://github.com/kubeflow/community/blob/master/CODE_OF_CONDUCT.md)** - All contributors must follow these guidelines
+2. **Fork and clone** the repository
+3. **Set up environment** - see [ONBOARDING.md](ONBOARDING.md)
+4. **Create feature branch** - `git checkout -b component/my-component`
+5. **Develop component** following the structure below
+6. **Test thoroughly** - see [TESTING.md](TESTING.md)
+7. **Submit pull request**
+
+## What We Accept
+
+*Types of contributions we welcome to the repository.*
+
+- **Individual Components**: ML tasks (preprocessing, training, evaluation)
+- **Component Collections**: Related component sets
+- **Pipeline Templates**: Complete pipeline examples
+- **Bug Fixes**: Improvements to existing components
+
+## Component Structure
+
+*Standard directory layout and files required for every component.*
+
+Each component must follow this structure:
```

Comment:
we have already mentioned this in onboarding right? could we instead link to the onboarding doc instead of repeating it here? Or we can just say in general "read the onboarding doc for getting started"